src/mapgen.cpp

```diff
@@ -451,15 +465,26 @@ static void GENERATOR_riot_damage( map &md, const tripoint_abs_omt &p )
             int behavior_roll = rng( 1, 100 );
 
             if( behavior_roll <= 20 ) {
-                md.add_field( current_tile, field_fd_blood );
+                if( tile_can_have_blood( md, current_tile, md.has_flag_ter( ter_furn_flag::TFLAG_WALL,
+                                         current_tile ), days_since_cataclysm ) ) {
+                    md.add_field( current_tile, field_fd_blood );
+                }
             } else if( behavior_roll <= 60 ) {
-                place_blood_streaks( md, current_tile );
+                place_blood_streaks( md, current_tile, days_since_cataclysm );
             } else {
-                place_bool_pools( md, current_tile );
+                place_bool_pools( md, current_tile, days_since_cataclysm );
             }
         }
-        if( x_in_y( 1, 2000 ) ) {
-            md.add_field( current_tile, field_fd_fire );
+
+        // Decrease the chance of fire spawning as the Cataclysm progresses to the min fire chance of 1 in 10,000
+        if( x_in_y( 1,  std::min( 2000 + 714 * days_since_cataclysm, 10000 ) ) ) {
+            if( md.has_flag_ter_or_furn( ter_furn_flag::TFLAG_FLAMMABLE, current_tile ) ||
+                md.has_flag_ter_or_furn( ter_furn_flag::TFLAG_FLAMMABLE_ASH, current_tile ) ||
+                md.has_flag_ter_or_furn( ter_furn_flag:: TFLAG_FLAMMABLE_HARD, current_tile ) ||
+                days_since_cataclysm < 3 ) {
+                // Only place fire on flammable surfaces unless the cataclysm started very recently
```

Comment:
This check presently doesn't really do anything because our floors are `FLAMMABLE`, and floors are 80% of the map (the rest of it being flammable wooden furniture).
But if we exclude floors from this check, then - combined with the x_in_y tweak above - the chance of fire would probably drop *too* much, and more than you intend.

I don't know a good solution here, just noting that this code *looks* buggy.

If you're satisfied with how this works in practice, then please add a comment noting that this is intentional. Something like
```suggestion
                // Only place fire on flammable surfaces unless the cataclysm started very recently.
                // Note that floors are usually FLAMMABLE_HARD
```

Reply:
It doesn't do a whole lot: strictly speaking it doesn't limit the possible spawnable fire terrain by much. But that's not exactly the intention behind it. And the comment could be adjusted to better reflect that.

The idea is right after the Cataclysm you might find fire pretty much anywhere. Just lingering street fires, bush fires, just fires wherever. The riots were a mess. But as time goes on I want to transition to fires only really popping up in places that make a bit more since. Locations that are actually flammable like inside houses.

This check just stops fires from appearing say, in the middle of the road or parking lot, or some random patch of grass, Concrete floor. ect.. It's not huge, just a minor thing to make things feel less hectic after awhile. And try to make fires feel less spontaneous like they just manifested themselves, and more grounded like they emerged from (to me)  logical sources.

I will adjust my comment to better reflect my intention.